### DISCUSSION TITLE
grid_distance not working in several cases

### DISCUSSION THREAD

**Discussion by heddha** [uber/h3#726] [Q&A]
Hey there everyone :)

I really like working with the H3 indexing system, but I ran into a problem quite a few times now, and not only in one resolution:

In various cases, grid_distance(h3id_1, h3id_2), doesn't work, while a circumference search of the specific id (e.g. "h3id_2 in grid_disk(h3id_1, x)" ) evaluates to true (just as an example showing that both are valid h3 ids that should actually return a valid distance).

I'm really not sure what could cause this, but really like working with h3 in my project. Is there any chance you could find out (and hopefully fix) the reason for this?

Thank you so much in advance!

Cheers,

heddha

Here are some tuples on which the comparison doesn't work:
820817fffffffff 82082ffffffffff
82195ffffffffff 823e67fffffffff
82195ffffffffff 823e6ffffffffff
82195ffffffffff 822db7fffffffff
82195ffffffffff 822db7fffffffff
82195ffffffffff 822db7fffffffff

83195dfffffffff 832da2fffffffff
83195dfffffffff 833f41fffffffff
83195dfffffffff 832da6fffffffff
83195dfffffffff 833f4dfffffffff

84195d1ffffffff 842db15ffffffff
84195d1ffffffff 842db1dffffffff
84195d1ffffffff 842db4bffffffff

85081337fffffff 8508047bfffffff
85080077fffffff 8508061bfffffff
85080653fffffff 850803cbfffffff
85081507fffffff 85082b13fffffff
85195d03fffffff 853e6123fffffff

86081335fffffff 86080479fffffff
860800667ffffff 86080619fffffff
86080651fffffff 860803c9fffffff
860815057ffffff 86082b10fffffff
86196919fffffff 863e6131fffffff

87081335bffffff 870804799ffffff
870800664ffffff 87080619dffffff
870806518ffffff 870803c9cffffff
870815055ffffff 87082b11dffffff
87196919affffff 873e6131bffffff
87196919affffff 873e6ca2bffffff

88081335b7fffff 8808047999fffff
8808006647fffff 88080619d9fffff
8808065189fffff 880803c9c5fffff
8808150551fffff 88082b11dbfffff
88196919abfffff 883e6131b3fffff

8908132743bffff 89080479997ffff
89080075d9bffff 89080619d9bffff
89080651887ffff 890803c9c5bffff
89081505503ffff 89082b11db3ffff
891e0375467ffff 89356da0aabffff

**Comment by nrabinowitz** [comment]
This is a known limitation of `gridPathCells`, and [documented here](https://h3geo.org/docs/api/traversal#gridpathcells):

> This function may fail to find the line between two indexes, for example if they are very far apart. It may also fail when finding distances for indexes on opposite sides of a pentagon.

<img width="301" alt="image" src="https://user-images.githubusercontent.com/163940/200392551-21a61130-e6f8-47b9-ac6f-7aeafff826a8.png">

This is something we expect to be able to fix, but we don't have a timeline for this at present.
